### PR TITLE
Ability to give other people points

### PR DESCRIPTION
**Pull request by tanndlin**
Allows users to `!give @Tanndlin all` or something like that
I'm going to wait to merge until after #732 is merged to simplify that PR

### CODE REVIEW

**Comment by tanndlin** [comment]
This was supposed to be a draft. I'm going to wait to merge until after #732 is merged to simplify that PR

File: src/app/plugins/give.plugin.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      return false;
+    }
+
+    return message.mentions.users.size === 1 && this.commandPattern.test(args[1]);
+  }
+
+  public async execute(message: IMessage, args: string[]) {
+    const userDoc = await this.container.pointService.getUserPointDoc(message.author.id);
+    const recipientUser = message.mentions.users.first()!; // Asserting true after validate
+    const recipient = await this.container.pointService.getUserPointDoc(recipientUser.id);
```
Comment (suneettipirneni) [inline src/app/plugins/give.plugin.ts:33]:
```suggestion
    const recipientUser = message.mentions.users.first()!; // Asserting true after validate
    const [userDoc, recipient] = await Promise.all([message.author.id, recipientUser.id].map(id => this.container.pointService.getUserPointDoc(id)));
```
These async tasks can run concurrently.

File: src/app/plugins/give.plugin.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    recipient: PointsDocument,
+    amount: number
+  ): Promise<MessageEmbed> {
+    await Promise.all([
+      this.container.pointService.awardPoints(giver.userID, -amount),
+      this.container.pointService.awardPoints(recipient.userID, amount),
+    ]);
+
+    const giverUser = await this.container.clientService.users.fetch(giver.userID);
+    const recipentUser = await this.container.clientService.users.fetch(recipient.userID);
```
Comment (suneettipirneni) [inline src/app/plugins/give.plugin.ts:59]:
```suggestion
    const [giverUser, recipientUser] = await Promise.all([giver.userID, recipient.userID].map(id => this.container.clientService.users.fetch(id));
```
Same here

**Review by suneettipirneni — CHANGES_REQUESTED** [review]

**Comment by Armintron** [comment]
Is this being picked back up?

**Comment by tanndlin** [comment]
> Is this being picked back up?

It's been like 6 hours 😭

**Comment by Armintron** [comment]
> > Is this being picked back up?
> 
> It's been like 6 hours 😭

Lmao my bad, just knew that it was put in a waiting state for the other pr

**Comment by tanndlin** [comment]
> Lmao my bad, just knew that it was put in a waiting state for the other pr

You good lol

File: src/app/plugins/give.plugin.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  public override pluginChannelName: string = Constants.Channels.Public.Games;
+
+  public override commandPattern: RegExp = /(all|\d+)/;
+
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public override validate(message: IMessage, args: string[]): boolean {
+    if (args.length !== 2) {
```
Comment (c650) [inline src/app/plugins/give.plugin.ts:23]: Are args correct if the mention is of a multi word name? Can you test that please?

if they don’t do the right thing let’s consider stripping args of all the mentions

File: src/app/plugins/give.plugin.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  public async execute(message: IMessage, args: string[]) {
+    const recipientUser = message.mentions.users.first()!; // Asserting true after validate
+    const [userDoc, recipient] = await Promise.all(
+      [message.author.id, recipientUser.id].map((id) =>
+        this.container.pointService.getUserPointDoc(id)
+      )
+    );
+
+    // Try to gamble the number given by user
+    const pointsToGive = args[1].toLowerCase() === 'all' ? userDoc.numPoints : parseInt(args[1]);
```
Comment (c650) [inline src/app/plugins/give.plugin.ts:39]: Does parseInt ever throw? I can’t remember

File: src/app/plugins/give.plugin.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+      embeds: [await this._give(userDoc, recipient, pointsToGive)],
+    });
+  }
+
+  private async _give(
+    giver: PointsDocument,
+    recipient: PointsDocument,
+    amount: number
+  ): Promise<MessageEmbed> {
+    console.log(amount);
```
Comment (c650) [inline src/app/plugins/give.plugin.ts:56]: Nix

File: src/app/plugins/give.plugin.ts (hunk 3 above)
Comment (tanndlin) [reply src/app/plugins/give.plugin.ts:23]: It uses mentions, which in plain text is <@1111111111111> no spaces
If it doesn't follow that form its not a ping

File: src/app/plugins/give.plugin.ts (hunk 4 above)
Comment (tanndlin) [reply src/app/plugins/give.plugin.ts:39]: It returns `NaN` but I test that arg with a regex for only digits in `validate`

**Pull request merged**